go-migrate/migrate.go

```diff
@@ -29,6 +29,13 @@ type Migration interface {
 	Revert(Options) error
 }
 
+type SupportNoRevert interface {
```

Comment:
Ah, hrm...  Maybe then a simple array of booleans for checking which versions are supported would be the cleanest solution here